lib/pbench/test/unit/agent/test_tool_meister.py

```diff
@@ -168,46 +169,39 @@ def f(directory: pathlib.Path, tar_file: pathlib.Path):
                 args=[], returncode=returncode, stdout=stdout, stderr=None
             )
 
+        TestSendDirectory.function_called.append("fake_tar")
         return f
 
-    @responses.activate
-    def test_tar_create_success(self, agent_setup, monkeypatch, logger_fixture):
-        """This test should pass the tar creation in send directory"""
-
-        directory = agent_setup["tmp"] / f"{TestCreateTar.tm_params['hostname']}"
-        function_called = 0
-
-        def fake_rmtree(directory: pathlib.Path):
-            assert directory == agent_setup["tmp"]
-            nonlocal function_called
-            function_called += 1
-            pass
-
-        monkeypatch.setattr(shutil, "rmtree", fake_rmtree)
-
-        def fake_md5(tar_file: pathlib.Path):
-            assert tar_file == pathlib.Path(f"{directory}.tar.xz")
-            nonlocal function_called
-            function_called += 1
-            return 10, "random_md5"
+    @staticmethod
+    def fake_unlink(*args):
+        assert args[0] == pathlib.Path(f"{TestSendDirectory.directory}.tar.xz")
+        TestSendDirectory.function_called.append("fake_unlink")
+        pass
 
-        monkeypatch.setattr("pbench.agent.tool_meister.md5sum", fake_md5)
+    @staticmethod
+    def fake_open(*args):
+        assert args[0] == pathlib.Path(f"{TestSendDirectory.directory}.tar.xz")
+        TestSendDirectory.function_called.append("fake_open")
+        return io.StringIO()
 
-        def fake_unlink(*args):
-            assert args[0] == pathlib.Path(f"{directory}.tar.xz")
-            nonlocal function_called
-            function_called += 1
-            pass
+    def fake_rmtree(self, directory: pathlib.Path):
+        assert directory == tmp_dir
+        self.function_called.append("fake_rmtree")
+        pass
 
-        monkeypatch.setattr(pathlib.Path, "unlink", fake_unlink)
+    def fake_md5(self, tar_file: pathlib.Path):
+        assert tar_file == pathlib.Path(f"{self.directory}.tar.xz")
+        self.function_called.append("fake_md5")
+        return 10, "random_md5"
```

Comment:
I believe the following hypothesis is happening:
As I mentioned above `self` is bound to a `TestSendDirectory` object, when the source code calls the `md5sum` function, the test refers to whatever is patched by the `monkeypatch` (in this case its a `fake_md5` instance method of `TestSendDirectory` object), this instance method also takes the same arguments as the original method so that works.

Reply:
See if https://github.com/npalaska/pbench/pull/2 addresses this satisfactorily.